ffi/tests/transparent.rs

```diff
@@ -31,6 +31,33 @@ pub struct TransparentStruct {
     _zst3: PhantomData<String>,
 }
```

Comment:
Deriving `FfiType` for a transparent item is problematic because either:
1. the user has to write a validation function
2. the user must mark the type as robust 

This API is bad for the following reasons:
* devs often don't understand what `#[ffi_type(unsafe {robust})]` does
* `#[ffi_type(unsafe {robust})]` implements unsafe trait in a macro so the user might be unaware of the safety implications. This is visually mitigated with the use of `unsafe {robust}` in the attribute

When validation function is required the user has to write something like the following impl (taken from this module):
```rs
iroha_ffi::ffi_type! {
    unsafe impl Transparent for NonRobustTransparent {
        type Target = NonRobustTransparentInner;

        validation_fn=unsafe {|target| target != &[0; 4]},
        niche_value=[0; 4]
    }
}
```

I could also require the use to write an impl like this in the robust case instead of using `ffi_type(unsafe {robust})` like so:
```rs
iroha_ffi::ffi_type! {
    unsafe impl Transparent for TransparentStruct {
        type Target = GenericTransparentStruct<()>;
    }
}
```

Reply:
As we disscussed in DM some days ago, we can consider approach with unsafe marker trait:

```rust
#[derive(SomeUnsafeFfiStuff)] // Won't compile without the following impl
pub struct MyStruct { ... }

// SAFETY: I've read the docs, I know what I'm doing
unsafe impl MeaningfulFfiMarker for MyStruct {}
```

Reply:
IMO having a few example impls outside the macro context is a good idea. While the blanket impl could also serve that purpose and the marker trait(s) could indicate different varieties of robust types. 

I do agree that the documentation on FFI needs to be expanded. With luck, we should be getting another tech writer, and should have a thoroughly documented FFI.